### ISSUE TITLE
[Bug] npu_dequant_swiglu_quant accuracy

### ISSUE THREAD

**Issue by SYChen123** [sglang-npu/sglang_npu#31]
### Checklist

- [ ] 1. I have searched related issues but cannot get the expected help.
- [ ] 2. The bug has not been fixed in the latest version.
- [ ] 3. Please note that if the bug-related issue you submitted lacks corresponding environment info and a minimal reproducible demo, it will be challenging for us to reproduce and resolve the issue, reducing the likelihood of receiving feedback.
- [ ] 4. If the issue you raised is not a bug but a question, please raise a discussion at https://github.com/sgl-project/sglang/discussions/new/choose Otherwise, it will be closed.
- [ ] 5. Please use English, otherwise it will be closed.

### Describe the bug

基线输出的token
<img width="1262" height="280" alt="Image" src="https://github.com/user-attachments/assets/cc9f9322-33af-4b20-86c5-1e969df5545d" />

使用npu_dequant_swiglu_quant输出的token，虽然有精度但和基线不完全一致
<img width="1248" height="235" alt="Image" src="https://github.com/user-attachments/assets/a719462d-c436-40c8-9bd7-b12a9f0b94ad" />

测试npu_grouped_matmul + npu_dequant_swiglu_quant输出，并将npu_grouped_matmul + npu_swiglu + npu_dynamic_quant单算子输出作为golden进行对比，发现二者输出也不完全相同。

### Reproduction

**P节点**
USE_VLLM_CUSTOM_ALLREDUCE=1 python -m sglang.launch_server --model-path /home/fuyong/codes/sglang/weights/ --disaggregation-ib-device mlx5_0 --disaggregation-mode prefill --host 141.61.105.146 --port 8000 --trust-remote-code --dist-init-addr 141.61.105.146:5000 --nnodes 1 --node-rank 0 --tp-size 16 --dp-size 1 --mem-fraction-static 0.8 --attention-backend ascend --device npu --quantization w8a8_int8 --disable-overlap-schedule --disaggregation-transfer-backend ascend

**D节点**
USE_VLLM_CUSTOM_ALLREDUCE=1 python -m sglang.launch_server --model-path /home/fuyong/codes/sglang/weights/ --disaggregation-ib-device mlx5_0,mlx5_1 --disaggregation-mode decode --host 141.61.105.147 --port 8001 --trust-remote-code --dist-init-addr 141.61.105.147:5000 --nnodes 2 --node-rank 0 --tp-size 32 --dp-size 32 --mem-fraction-static 0.8 --max-running-requests 48 --attention-backend ascend --device npu --quantization w8a8_int8 --enable-deepep-moe --enable-dp-attention --deepep-mode low_latency --enable-dp-lm-head --moe-dense-tp-size 1 --disaggregation-transfer-backend ascend

USE_VLLM_CUSTOM_ALLREDUCE=1 python -m sglang.launch_server --model-path /home/fuyong/codes/sglang/weights/ --disaggregation-ib-device mlx5_0,mlx5_1 --disaggregation-mode decode --host 141.61.105.148 --port 8001 --trust-remote-code --dist-init-addr 141.61.105.147:5000 --nnodes 2 --node-rank 1 --tp-size 32 --dp-size 32 --mem-fraction-static 0.8 --max-running-requests 48 --attention-backend ascend --device npu --quantization w8a8_int8 --enable-deepep-moe --enable-dp-attention --deepep-mode low_latency --enable-dp-lm-head --moe-dense-tp-size 1 --disaggregation-transfer-backend ascend

### Environment

PD分离，D节点双机跑满层deepseek模型

**Comment by SYChen123** [comment]
**问题结论**
1. 接入npu_dequant_swiglu_quant算子前，用的是npu_grouped_matmul + npu_swiglu + npu_dynamic_quant实现，其中npu_dynamic_quant无法做分组量化，只能做per-token量化，量化参数通过取每一行的最大值rowMax计算得到的。如下图所示。
<img width="242" height="44" alt="Image" src="https://github.com/user-attachments/assets/afe22f92-1829-429f-a5b0-88de44b0cdcc" />

2. 接入npu_dequant_swiglu_quant算子后，我们向该算子传入group_list做分组量化，量化参数是通过取每一个组内的最大值计算得到的。

因此，这两种方案量化时使用的量化参数不同，行为是有区别的，因此输出的logits无法完全一致是合理的。
